Core/HistoryManager.swift

```diff
@@ -17,21 +17,182 @@
 //  limitations under the License.
 //
 
+import CoreData
 import Foundation
 import BrowserServicesKit
+import History
+import Common
+import Persistence
 
-class HistoryManager {
+public protocol HistoryManaging {
 
-    let privacyConfig: PrivacyConfiguration
+    var historyCoordinator: HistoryCoordinating { get }
+
+}
+
+public class HistoryManager: HistoryManaging {
+
+    let privacyConfigManager: PrivacyConfigurationManaging
     let variantManager: VariantManager
+    let database: CoreDataDatabase
+
+    private var currentHistoryCoordinator: HistoryCoordinating?
+
+    public var historyCoordinator: HistoryCoordinating {
+        guard isHistoryFeatureEnabled() else {
+            currentHistoryCoordinator = nil
+            return NullHistoryCoordinator()
+        }
+
+        if let currentHistoryCoordinator {
+            return currentHistoryCoordinator
+        }
+
+        database.loadStore()
 
-    init(privacyConfig: PrivacyConfiguration, variantManager: VariantManager) {
-        self.privacyConfig = privacyConfig
+        let context = database.makeContext(concurrencyType: .privateQueueConcurrencyType)
+        let historyCoordinator = HistoryCoordinator(historyStoring: HistoryStore(context: context, eventMapper: HistoryStoreEventMapper()))
+        currentHistoryCoordinator = historyCoordinator
+        historyCoordinator.loadHistory {
+            // no-op
+        }
+        return historyCoordinator
+    }
+
+    public init(privacyConfigManager: PrivacyConfigurationManaging, variantManager: VariantManager, database: CoreDataDatabase) {
+        self.privacyConfigManager = privacyConfigManager
         self.variantManager = variantManager
+        self.database = database
     }
 
     func isHistoryFeatureEnabled() -> Bool {
-        return privacyConfig.isEnabled(featureKey: .history) && variantManager.isSupported(feature: .history)
+        return privacyConfigManager.privacyConfig.isEnabled(featureKey: .history) && variantManager.isSupported(feature: .history)
+    }
+
+    public func removeAllHistory() async {
+        await withCheckedContinuation { continuation in
+            historyCoordinator.burnAll {
+                continuation.resume()
+            }
+        }
+    }
+
+}
+
+class NullHistoryCoordinator: HistoryCoordinating {
```

Comment:
Basically a no-op history coordinator for use when history is disabled and in the tab previews.